tests/samplers_tests/test_grid.py

```diff
@@ -228,3 +228,10 @@ def test_enqueued_insufficient_trial() -> None:
 
     with pytest.raises(ValueError):
         study.optimize(lambda trial: trial.suggest_int("a", 0, 100))
+
+
+def test_nan() -> None:
+    sampler = optuna.samplers.GridSampler({"x": [0, float("nan")]})
+    study = optuna.create_study(sampler=sampler)
+    study.optimize(lambda _: 0)
```

Comment:
nit: How about sampling "x" as this test case is for sampler.

```suggestion
    study.optimize(lambda trial: trial.suggest_float("x", 0, 1))
```

Reply:
You cannot return `nan` as a value.

Reply:
Thanks, I'll fix it.